ingestion/src/metadata/ingestion/source/database/databricks/metadata.py

```diff
@@ -661,3 +663,38 @@ def get_table_description(
                 f"Table description error for table [{schema_name}.{table_name}]: {exc}"
             )
         return description
+
+    def _filter_owner_name(self, owner_name: str) -> str:
+        """remove unnecessary keyword from name"""
+        pattern = r"\(Unknown\)"
+        filtered_name = re.sub(pattern, "", owner_name).strip()
+        return filtered_name
+
+    def _check_if_email(self, email_id: str) -> bool:
+        """check if email string is valid"""
+        email_pattern = r"^[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$"
```

Comment:
do we need to do this here? Seems like a more generic check rather than keeping it only for databricks. How are we solving it for other systems such as postgres where usernames are not necessarily emails?

Reply:
In the databricks, we get owner name in both name/email type. So that I have applied here to filter out based on email first and then name.
While in other db/dashboard connection we have confirm that we get owner's email(from api/inspector method) and we're passing directly to get reference.
[domodb](https://github.com/open-metadata/OpenMetadata/blob/b84ce33b80cb43e885540a928f439a75771acbf6/ingestion/src/metadata/ingestion/source/database/domodatabase/metadata.py#L168), [iceberg](https://github.com/open-metadata/OpenMetadata/blob/373cafcda26b17108d0897c54773df97e271447e/ingestion/src/metadata/ingestion/source/database/iceberg/metadata.py#L231), [metabase](https://github.com/open-metadata/OpenMetadata/blob/c84baaf66a23da75f54d126b807fca2d2682b9e1/ingestion/src/metadata/ingestion/source/dashboard/metabase/metadata.py#L154).
I think we can put this method in `DatabaseServiceSource` or `DashboardServiceSource` & check with valid email whenever we're not sure. WDYT ?

Reply:
- let's put this method in a common place. maybe https://github.com/open-metadata/OpenMetadata/blob/main/ingestion/src/metadata/utils/helpers.py? I don't love it but we don't have a better structure atm.
- let's try to use pydantic [emailStr](https://docs.pydantic.dev/1.10/usage/types/#:~:text=requires%20email%2Dvalidator%20to%20be%20installed%3B%20the%20input%20string%20must%20be%20a%20valid%20email%20address%2C%20and%20the%20output%20is%20a%20simple%20string) instead of regex